src/main/kotlin/org/rust/lang/core/psi/RsPsiFactory.kt

```diff
@@ -59,8 +59,17 @@ class RsPsiFactory(private val project: Project) {
         return path
     }
 
-    fun createStructLiteralField(name: String): RsStructLiteralField =
-        createExpressionOfType<RsStructLiteral>("S { $name: () }").structLiteralBody.structLiteralFieldList[0]
+    fun createStructLiteral(name: String): RsStructLiteral =
+        createExpressionOfType<RsStructLiteral>("$name { }")
+
+    fun createStructLiteralField(name: String, value: RsExpr? = null): RsStructLiteralField {
+        val structLiteralField = createExpressionOfType<RsStructLiteral>("S { $name: () }")
+            .structLiteralBody
+            .structLiteralFieldList[0]
+        if (value != null)
+            structLiteralField.expr?.replace(value)
```

Comment:
For short braceless ifs we usually place condition and action on the same line, to avoid gotofail bug:

```
if (value != null) structLIteralField.expr?.replace(value)
```